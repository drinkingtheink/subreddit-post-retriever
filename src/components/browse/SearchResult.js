import React, { Component } from 'react';

class SearchResultCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      headline: 'Pretty Cool, Right?'
    };
  }

  render() {

    let thumbStyle = {
      backgroundImage: 'url(' + this.props.result.thumbnail + ')'
    };

    function Thumbnail({thumbnail}) {
      if (thumbnail !== 'self') {
        return (
          <span className="post-thumbnail" style={thumbStyle}></span>
        )
      } else {
        return null
      }
    }

    function Author({ author, thumbnail }) {
      if (author) {
        return (
          <p className="post-author attributed">
            <Thumbnail thumbnail={thumbnail} /> 
            {author}
          </p>
        );
      } else {
        return (
          <p className="post-author unattributed">
            <i class="fa fa-user"></i> 
            Unknown
          </p>
        );
      }
    }

    return (
      <div className="search-result">
        <h2 className="post-headline">
          <a 
            href={this.props.redditBaseUrl + '/' + this.props.result.permalink} 
            alt={this.props.result.title}
            target="_blank"
            rel="noopener noreferrer"
            >
              {this.props.result.title}
          </a>
        </h2>
        <p className="result-display_name">{this.props.result.display_name_prefixed}</p>
        <Author author={this.props.result.author} thumbnail={this.props.result.thumbnail} />
      </div>
    )
  }
}

export default SearchResultCard;