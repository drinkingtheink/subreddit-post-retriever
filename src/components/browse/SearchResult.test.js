import React from 'react'
import ReactDOM from 'react-dom'
import SearchResult from './SearchResult'
import TestRenderer from 'react-test-renderer'
import { exampleSearchResult } from './test/exampleSearchResult.js'

const compName = 'SearchResult'
const testRenderer = TestRenderer.create(<SearchResult result={ exampleSearchResult } />);
const testInstance = testRenderer.root;

describe(`${compName} =====================`, () => {

  it(`renders without crashing`, () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchResult result={ exampleSearchResult } />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

});