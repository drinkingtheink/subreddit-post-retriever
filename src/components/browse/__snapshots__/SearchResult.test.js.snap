// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult ===================== renders correctly with no props 1`] = `ShallowWrapper {}`;
